### PR TITLE
Add new console command "cache:clear:all" using shopware CacheClearer component

### PR DESCRIPTION
**Pull request by raffaelecarelle**
Implemented the KernelCacheConfigCompilerPass to customize cache clearing behavior using the SymfonyCacheClearerAdapter (wrapper for CacheClearer Shopware component). Added a unit test to validate the replacement of the default cache clearer with the adapter and ensure no tag-related issues arise during the process.

This change alowed to clear container cache, old folders cache and much more throuth the cache:clear symfony command.

### 5. Checklist

- [ ] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfill them.

### CODE REVIEW

**Comment by aragon999** [comment]
This should get a changelog :-)

**Comment by shyim** [comment]
With this change also the object cache would be cleared right? I don't know if we want to have this 🤔

**Comment by raffaelecarelle** [comment]
> With this change also the object cache would be cleared right? I don't know if we want to have this 🤔

The behaviur is the same of Clear Cache from che Administration > Settings > Cache & Indexes of the Backoffice.

I think this is useful because during development is more complicated go to that page but it is more productive and faster use cache:clear command.

**Comment by ghost** [comment]
> With this change also the object cache would be cleared right? I don't know if we want to have this 🤔

I also think that the pools should not be flushed by "cache:clear".

> I think this is useful because during development is more complicated go to that page but it is more productive and faster use cache:clear command.

You can already use the command "cache:pool:clear --all" to achieve this.
I think it is good to have the two explicit commands "cache:clear" and "cache:pool:clear".

File: src/Core/Framework/Adapter/Cache/SymfonyCacheClearerAdapter.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class SymfonyCacheClearerAdapter implements CacheClearerInterface
+{
+    public function __construct(
+        private readonly ShopwareCacheClearer $cacheClearer
+    ) {
+    }
+
+    public function clear(string $cacheDir): void
+    {
+        $this->cacheClearer->clear();
```
Comment (ghost) [inline src/Core/Framework/Adapter/Cache/SymfonyCacheClearerAdapter.php]: You could add a optinal parameter like "$clearAdapterCache" to the ShopwareCacheClearer and set it to "false" here.

**Comment by raffaelecarelle** [comment]
> > With this change also the object cache would be cleared right? I don't know if we want to have this 🤔
> 
> I also think that the pools should not be flushed by "cache:clear".
> 
> > I think this is useful because during development is more complicated go to that page but it is more productive and faster use cache:clear command.
> 
> You can already use the command "cache:pool:clear --all" to achieve this. I think it is good to have the two explicit commands "cache:clear" and "cache:pool:clear".

it's fine to have two separated command, but it is also be good to have old container directories clear task

File: src/Core/Framework/Adapter/Cache/SymfonyCacheClearerAdapter.php (hunk 1 above)
Comment (raffaelecarelle) [reply src/Core/Framework/Adapter/Cache/SymfonyCacheClearerAdapter.php]: Ok, I instroduce this parameter to prevent adapter cache to be cleared.

**Comment by AydinHassan** [comment]
Hey @raffaelecarelle thanks for the PR but I think we would like to keep these commands as default Symfony. We could introduce a new one, maybe `cache:clear:all` that does everything? What do you think?

**Comment by raffaelecarelle** [comment]
> Hey @raffaelecarelle thanks for the PR but I think we would like to keep these commands as default Symfony. We could introduce a new one, maybe `cache:clear:all` that does everything? What do you think?

Hello @AydinHassan, yeah this is another solution I thought me too. would you like that I develop it in this PR or another one?

**Comment by AydinHassan** [comment]
@raffaelecarelle this one is fine :)

**Pull request closed without merging**